packages/arb-token-bridge-ui/src/pages/api/deposits.ts

```diff
@@ -0,0 +1,108 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { gql } from '@apollo/client'
+import {
+  FetchDepositsFromSubgraphResult,
+  getL1SubgraphClient
+} from 'token-bridge-sdk'
+
+// Extending the standard NextJs request with Deposit-params
+type NextApiRequestWithDepositParams = NextApiRequest & {
+  query: {
+    l2ChainId: string
+    address: string
+    search?: string
+    page?: string
+    pageSize?: string
+    fromBlock?: string
+    toBlock?: string
+  }
+}
+
+type DepositsResponse = {
+  data: FetchDepositsFromSubgraphResult[]
+  message?: string // in case of any error
+}
+
+export default async function handler(
+  req: NextApiRequestWithDepositParams,
+  res: NextApiResponse<DepositsResponse>
+) {
+  try {
+    const {
+      address,
+      search = '',
+      l2ChainId,
+      page = '0',
+      pageSize = '10',
+      fromBlock,
+      toBlock
+    } = req.query
+
```

Comment:
we should also validate if the http method is `GET` (see the screening endpoint)

Reply:
Done